File: Pre CAT/DS/InsertionSort.c
#include<stdio.h>
void insertionSort(int* arr,int size){
    for(int i=1;i<size;i++){
        int key = arr[i];
        int j;
        for(j=i-1;j>=0;j--){
            if(arr[j]<key){
                break;
            }
            arr[j+1]=arr[j];
        }
        arr[j+1]=key;
    }
}
int main(){
     int size;
    printf("Enter the size of array: ");
    scanf("%d",&size);
    int arr[size];
    printf("Enter the elements of array: ");
    for(int i=0;i<size;i++){
         scanf("%d",&arr[i]);
    }
    printf("Before Sort:");
    for(int i=0;i<size;i++){
        printf("%d ",arr[i]);
    }
    insertionSort(arr,size);
    printf("\nAfter Sort: ");
    for(int i=0;i<size;i++){
        printf("%d ",arr[i]);
    }
}

File: CDAC/edac_data_structure/src/edac_data_structure/InsertionSort.java
package edac_data_structure;

import java.util.Scanner;

public class InsertionSort {
	public static void insertionSort(int[] arr) {
		for(int i=1;i<arr.length;i++) {
			int key = arr[i];
			int j=i-1;
			while(j>=0 &&arr[j]>key) {
				arr[j+1]=arr[j];
				j--;
			}
			arr[j+1]=key;
		}
	}
	
	public static void main(String[] args) {
		System.out.println("Enter the size of array: ");
		Scanner sc = new Scanner(System.in);
		int size=sc.nextInt();
		int arr[]=new int[size];
		System.out.println("Enter the elements of the array: ");
		for(int i=0;i<arr.length;i++) {
			arr[i]=sc.nextInt();
		}
		insertionSort(arr);
		for(int i=0;i<arr.length;i++) {
			System.out.print(arr[i]+" ");
		}
		System.out.println();
		sc.close();
	}
}
